smalltalksrc/Slang/TMethod.class.st

```diff
@@ -651,9 +658,19 @@ TMethod >> declarationAt: aVariableName ifPresent: presentBlock [
 ]
 
 { #category : #accessing }
-TMethod >> declarationAt: aVariableName  "<String>" put: aDeclaration [ "<String>" "^aDeclaration"
+TMethod >> declarationAt: aVariableName put: aDeclaration [
+
+	| isRelevantVariable |
+	isRelevantVariable := (aVariableName beginsWithAnyOf:
+		                       #( #self , #cascade , #toDoLimit )) not.
 
-	^declarations at: aVariableName put: aDeclaration
+
+	isRelevantVariable and: [
+		(self declaresVariableWithName: aVariableName) ifFalse: [
+			TranslationError signal:
+				'Undefined local or argument ' , aVariableName
+				, ' type declaration' ] ].
```

Comment:
This `and:` is weird.

```suggestion
	(isRelevantVariable and: [
		self declaresVariableWithName: aVariableName]) ifFalse: [
			TranslationError signal:
				'Undefined local or argument ' , aVariableName
				, ' type declaration' ].
```

Reply:
We can move this logic to a `validateVariableDeclarationExists` method

Reply:
rewrote it as
```smalltalk
(shouldCheckVariable and: [
		 (self declaresVariableWithName: aVariableName) not]) ifTrue: [
		TranslationError signal:
			'Undefined local or argument ' , aVariableName
			, ' type declaration' ]
```